src/main/java/com/codejoust/main/service/GameManagementService.java

```diff
@@ -283,14 +307,131 @@ public GameDto manuallyEndGame(String roomId, EndGameRequest request) {
         return gameDto;
     }
 
-    protected void handleEndGame(Game game) {
+    public void handleEndGame(Game game) {
         // Cancel all previously scheduled timers
         GameTimer gameTimer = game.getGameTimer();
         gameTimer.getTimer().cancel();
 
         for (Timer timer : gameTimer.getNotificationTimers()) {
             timer.cancel();
         }
+
+        CreateGameReportTask createGameReportTask = new CreateGameReportTask(this, game);
+        Timer createGameReportTimer = new Timer();
+        createGameReportTimer.schedule(createGameReportTask, GameTimer.DURATION_1 * 1000);
+    }
+
+    public GameReport createGameReport(Game game) {
+        GameReport gameReport = new GameReport();
+        int numProblems = game.getProblems().size();
+        int numPlayers = game.getPlayers().size();
+
+        // Initialize the statistic variables for each problem.
+        int[] userSolved = new int[numProblems];
```

Comment:
In order to make this more readable, maybe split it into separate methods? Something like `generateGameStatistics`, `generateProblemStats`, `populateAccounts`, etc. that way it's more clear what each section is doing

Reply:
I split it up into `computeGameStatisticsAndAddSubmissions`, setGameReportStatistics, `createProblemContainers, and `addGameReportAccounts`. The first method is clearly complicated (given the `And` in the name), but it fits fairly naturally because in order to calculate those game statistics, I have to iterate through each user's submissions.

Let me know if there's another system that you think would be better.